web/src/tippyjs.ts

```diff
@@ -196,6 +196,20 @@ export function initialize(): void {
         },
     });
 
+    delegate("body", {
+        target: "#org-other-settings .subsection-header .save-button-controls .subsection-changes-save",
+        appendTo: () => document.body,
+        onShow(instance) {
+            const $elem = $(instance.reference);
+            if ($($elem).find(".save-button").is(":disabled")) {
+                const content = $t({defaultMessage: "Cannot save invalid Jitsi server URL."});
+                instance.setContent(content);
+                return undefined;
```

Comment:
This logic still doesn't have anything in its checks specific to the Jitsi URL input being disabled.

Reply:
Hi @timabbott
In the file web/src/settings_components.js, following the trace when the check if the save button should be disabled or not, the Jitsi URL is checked according to:
enable_or_disable_save_button() --> should_disable_save_button_for_jitsi_server_url_setting() --> util.is_valid_url() 

Here only the start of the url is checked, shouldn't the check of the Jitsi URL also be done here rather than inside of the tippy? In order for the tippy to be synchronized with the state (enabled/disabled) of the save button. We propose to change from util.is_valid_url() to a new method utils.is_valid_jitsi_url() that checks the URL using regex (since every Jitsi URL have the format, https://meet.jit.si + a room number) instead of adding a check to the tippy?

Reply:
The code probably works, but it's fragile and not something one would want to merge in a large project. 

Imagine if we added another situation where the save button would be disabled for some other reason, this logic would display a tooltip that the reason was an invalid Jitsi server URL, and it would not be obvious to someone working on the code for the setting that this would happen (because they're probably not going to look at this file at all while doing so). So it would be really good to, for example, call the function that validates the Jitsi server URL field before deciding to generate a tooltip that claims it is invalid.

Reply:
Thank you for the response @timabbott !  We understand that there will be a problem if, in the future, a save button is placed inside the element with this specific path "target: "#org-other-settings .subsection-header" as specified in the tippy. 

Regarding the validation function of the Jitsi server URL that you mentioned, we have tried to find it in the code base but can't seem to find the one you are pointing at. It would be helpful if you could give us a pointer of where to look. 

Lastly, are you proposing that we generate a tippy inside the "settings_components.js" where the save button is disabled/enabled or that we use the code we have suggested today, but add a check of the Jitsi URL?

Reply:
Would probably be better to just add the tooltip in the `settings_components.js` function which enables or disables the button. You can probably use `stream_ui_updates.initialize_disable_btn_hint_popover`.

Reply:
Hi @sahil839! Thanks for the help! We have now updated the code and now instead add the tippy in settings_components.js as you mentioned. However, we have a problem with circular dependencies which we do not know how to solve. We have looked the guidelines and the Zuilp chat but havn't found anything. Do you have an idea how to solve this?

Reply:
I think we can move `initialize_disable_btn_hint_popover` to `settings_components.js`. We anyways use that function in multiple places.

Reply:
@sahil839 Thanks for the response :)

We have moved `initialize_disable_btn_hint_popover` into `settings_components.js` and updated the code thereafter. In addition to this, we have created two commits, one with the move of the function and one with the new functionality for the url to make it easier for you to review. Let us know your feedback :)